### PR TITLE
refactor(Achievement): switch several Author reads to user_id

### PR DESCRIPTION
**Pull request by wescopeland**
This PR switches several `Achievement` table reads that rely on `Author` to instead query by `user_id`. There should be no functional changes in this PR.

There are several calls elsewhere in the codebase that still rely on the `Author` value. I'll tackle those in a separate PR.

### CODE REVIEW

File: app/Helpers/database/achievement.php [hunk 1]
```diff
@@ -27,13 +27,15 @@ function getAchievementsList(
...
-    $innerJoin = "";
+    $innerJoinPlayerAchievements = "";
     $withAwardedDate = "";
     if ($params > 0 && $user) {
-        $innerJoin = "LEFT JOIN player_achievements AS pa ON pa.achievement_id = ach.ID AND pa.user_id = " . $user->id;
+        $innerJoinPlayerAchievements = "LEFT JOIN player_achievements AS pa ON pa.achievement_id = ach.ID AND pa.user_id = " . $user->id;
         $withAwardedDate = ", COALESCE(pa.unlocked_hardcore_at, pa.unlocked_at) AS AwardedDate";
     }
 
+    $innerJoinUserAccounts = "LEFT JOIN UserAccounts AS ua ON ua.ID = ach.user_id";
```
Comment (Jamiras) [inline app/Helpers/database/achievement.php:37]: Why use a variable for this? There's other `INNER JOIN`s directly in the query.

File: resources/views/components/game/claim-info.blade.php [hunk 2]
```diff
@@ -72,7 +72,7 @@
         } else {
             // player_sessions only exist after 14 Oct 2023
             $achievement = Achievement::where('GameID', $gameId)
-                ->where('Author', $claim['User'])
+                ->where('user_id', $playerGame->user->id)
```
Comment (Jamiras) [inline resources/views/components/game/claim-info.blade.php:75]: This fails if the claimant has never played the game:
```
Attempt to read property "id" on null
```

File: app/Helpers/database/achievement.php (hunk 1 above)
Comment (wescopeland) [reply app/Helpers/database/achievement.php]: Good catch. This is leftover cruft from refactoring work. Fixed in latest.

File: resources/views/components/game/claim-info.blade.php (hunk 2 above)
Comment (wescopeland) [reply resources/views/components/game/claim-info.blade.php]: Thanks. Resolving this proved to be quite the rabbit hole.

The component currently suffers from an N+1 query problem in `master`. This is easily observable on game 20255.

I fixed in the error in latest, but I also rewrote nearly all of these queries to rely on eager-loaded model relationships in order to resolve the N+1 query issue.

File: resources/views/components/game/devbox-claim-management.blade.php [hunk 3]
```diff
@@ -37,11 +36,11 @@
 $userHasClaimSlot = false;
 $openTickets = null;
 
+$userPermissions = $user?->getAttribute('Permissions') ?? 0;
```
Comment (Jamiras) [inline resources/views/components/game/devbox-claim-management.blade.php:39]: Recommend using enum as fallback value: `Permissions::Unregistered`

Comment (wescopeland) [reply resources/views/components/game/devbox-claim-management.blade.php]: Agreed, done in latest.

**Review by Jamiras — APPROVED** [review]

**Pull request merged**